include/gatekeeper_gk.h

```diff
@@ -24,13 +24,116 @@
 #include "gatekeeper_ipip.h"
 #include "gatekeeper_ggu.h"
 #include "gatekeeper_mailbox.h"
+#include "gatekeeper_lpm.h"
+
+/* The maximum number of gateways that Gatekeeper may use. */
+#define GK_MAX_NUM_GW       (8)
+
+/*
+ * The LPM supports 8-bit next hop,
+ * which at most has 128 different values.
+ */
+#define GK_MAX_NUM_POLICIES (128)
 
 /*
  * A flow entry can be in one of three states:
  * request, granted, or declined.
  */
 enum gk_flow_state { GK_REQUEST, GK_GRANTED, GK_DECLINED };
 
+enum gk_policy_action {
+	/* Forward the packe to the corresponding Grantor. */
+	GK_FWD_GRANTOR,
+
+	/* Forward the packe to the corresponding gateway. */
+	GK_FWD_GATEWAY,
+
+	/*
+	 * The Grantor server is a neighbor.
+	 * Forward the packe to the Grantor server directly.
+	 */
+	GK_FWD_NEIGHBOR,
+
+	/* Forward the packet to the back interface. */
+	GK_FWD_BACK_NET,
+
+	/* Drop the packet. */
+	GK_DROP,
+
+	/* Invalid policy action. */
+	GK_POLICY_MAX,
+};
+
+struct gk_nexthop {
+
+	uint16_t proto;
+
+	union {
+		struct in_addr  v4;
+		struct in6_addr v6;
+	} ip;
+
+	/* Indicate whether the MAC address is stale or not. */
+	bool stale;
+
+	/* Cache the whole Ethernet header. */
+	struct ether_hdr eth_hdr;
+};
+
+struct gk_policy {
```

Comment:
Although the name "policy" here is correct, given that we already refer to policy what Grantor servers run to authorize flows, it's better to replace `policy` with `fib` (i.e. forward information base) here to avoid confusion.